src/main/java/io/confluent/idesidecar/restapi/proxy/ConnectionProcessor.java

```diff
@@ -51,6 +51,25 @@ public Future<T> process(T context) {
     // Store the connection details in the context
     context.setConnectionState(connectionState);
 
+    switch (context.getClusterType()) {
+      case KAFKA ->
+        connectionState
+            .getKafkaTLSConfig()
+            .ifPresent(kafkaOptions -> {
+              if (kafkaOptions.truststore() != null) {
+                context.setTruststoreOptions(kafkaOptions.truststore());
+              }
+            });
+      case SCHEMA_REGISTRY ->
+        connectionState
+            .getSchemaRegistryTLSConfig()
+            .ifPresent(schemaRegistryOptions -> {
+              if (schemaRegistryOptions.truststore() != null) {
+                context.setTruststoreOptions(schemaRegistryOptions.truststore());
+              }
+            });
+    }
```

Comment:
Do we need to provide the keystore in case of mTLS, too?

Reply:
Yes, but only for Schema Registry REST. We will think more about mTLS for Confluent Cloud as part of a follow up - #235